nibabel/spatialimages.py

```diff
@@ -461,8 +479,66 @@ def from_image(klass, img):
                      klass.header_class.from_header(img.header),
                      extra=img.extra.copy())
 
+    def slice_affine(self, idx):
```

Comment:
Should this be public?

Reply:
It seemed potentially useful, but I can make it private.